src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/Extensions/CastExpressionSyntaxExtensions.vb

```diff
@@ -10,21 +10,127 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
     Friend Module CastExpressionSyntaxExtensions
         <Extension>
         Public Function Uncast(cast As CastExpressionSyntax) As ExpressionSyntax
-            Return Uncast(cast, cast.Expression)
+            Return Uncast(cast, cast.OpenParenToken, cast.Expression, cast.CommaToken, cast.Type, cast.CloseParenToken)
         End Function
 
         <Extension>
         Public Function Uncast(cast As PredefinedCastExpressionSyntax) As ExpressionSyntax
-            Return Uncast(cast, cast.Expression)
+            Return Uncast(cast, cast.OpenParenToken, cast.Expression, Nothing, Nothing, cast.CloseParenToken)
         End Function
 
-        Private Function Uncast(castNode As ExpressionSyntax, innerNode As ExpressionSyntax) As ExpressionSyntax
-            Dim resultNode = innerNode.WithTriviaFrom(castNode)
+        Private Function Uncast(castNode As ExpressionSyntax, openParen As SyntaxToken, innerNode As ExpressionSyntax, commaToken As SyntaxToken, typeNode As TypeSyntax, closeParen As SyntaxToken) As ExpressionSyntax
+
+            Dim leadingTrivia As List(Of SyntaxTrivia) = castNode.GetLeadingTrivia.ToList
+            If openParen.LeadingTrivia.ContainsCommentOrLineContinue Then
+                leadingTrivia.AddRange(openParen.LeadingTrivia)
+            End If
+
+            If openParen.TrailingTrivia.ContainsCommentOrLineContinue Then
+                leadingTrivia.AddRange(openParen.TrailingTrivia)
+            End If
+            If (Not castNode.GetLeadingTrivia.Any) AndAlso leadingTrivia.FirstOrDefault.IsKind(SyntaxKind.WhitespaceTrivia) Then
+                leadingTrivia.RemoveAt(0)
+            End If
+
+            If innerNode.GetLeadingTrivia.ContainsCommentOrLineContinue Then
+                leadingTrivia.AddRange(innerNode.GetLeadingTrivia)
+            End If
+
+            Dim trailingTrivia As List(Of SyntaxTrivia) = castNode.GetTrailingTrivia.ToList
+            If closeParen.TrailingTrivia.ContainsCommentOrLineContinue Then
+                trailingTrivia.InsertRange(0, closeParen.TrailingTrivia)
+            End If
+
+            If closeParen.LeadingTrivia.ContainsCommentOrLineContinue Then
+                trailingTrivia.InsertRange(0, openParen.LeadingTrivia)
+            End If
+
+            If typeNode IsNot Nothing Then
```

Comment:
This will never be false

Reply:
It is for PredefinedCast where I pass in Nothing.
```
Public Function Uncast(cast As PredefinedCastExpressionSyntax) As ExpressionSyntax
    Return Uncast(cast, cast.OpenParenToken, cast.Expression, commaToken:=Nothing, typeNode:=Nothing, cast.CloseParenToken)
End Function
```